lua/entities/gmod_tardis/modules/sh_tracking.lua

```diff
@@ -0,0 +1,497 @@
+        if ply==pilot then
+            if down then
+                self:SetData("tracking-trace",true)
+            else
+                self:SendMessage("tracking-set", {self:GetData("tracking-ent")})
+            end
+        end
+        if not down then
+            self:SetData("tracking-trace",false)
+            self:SetData("tracking-ent",nil)
+        end
+    end,
+    key=KEY_X,
+    clientonly=true,
+    exterior=true
+})
+
+TARDIS:AddKeyBind("tracking-rotation",{
+    name="TrackRotation",
+    section="Flight",
+    func=function(self,down,ply)
+        local pilot = self:GetData("pilot")
+        if not self:GetTracking() or ply ~= pilot or not down then return end
+        self:ToggleTrackRotation()
+        TARDIS:StatusMessage(ply, "Controls.Tracking.Rotation", self:GetTrackRotation())
+    end,
+    key=KEY_T,
+    serveronly=true,
+    exterior=true
+})
+
+if SERVER then
+    function ENT:GetTracking()
+        return self:GetData("tracking-ent")
+    end
+
+    function ENT:SetTracking(ent,ply)
+        local wasTrackingEnt = self:GetData("tracking-ent")
+        local wasTracking = IsValid(wasTrackingEnt)
+        local isTracking = IsValid(ent)
+        local valid = true
+        if isTracking then 
+            if ent.TardisPart or ent.TardisInterior or (ent:IsPlayer() and IsValid(TARDIS:GetInteriorEnt(ent))) then
+                if IsValid(ply) then
+                    TARDIS:ErrorMessage(ply, "Controls.Tracking.InteriorFail")
+                end
+                valid = false
+            elseif ent == self then
+                if IsValid(ply) then
+                    TARDIS:ErrorMessage(ply, "Controls.Tracking.SelfFail")
+                end
+                valid = false
+            elseif ent:GetPos():Distance(self:GetPos()) > TRACKING_MAX_DISTANCE_SET then
+                if IsValid(ply) then
+                    TARDIS:ErrorMessage(ply, "Controls.Tracking.DistanceFail")
+                end
+                valid = false
+            elseif self:CallHook("CanTrack", ent, ply) == false then
+                if IsValid(ply) then
+                    TARDIS:ErrorMessage(ply, "Controls.Tracking.GenericFail")
+                end
+                valid = false
+            end
+            if not valid then
+                return false
+            end
+
+            local wasFlying = self:GetFlight()
+            if not wasFlying then
+                local success = self:SetFlight(true)
+                if not success then
+                    if IsValid(ply) then
+                        TARDIS:ErrorMessage(ply, "Controls.Tracking.FlightFail")
+                    end
+                    return false
+                end
+            end
+
+            if not wasTracking then
+                self:SetData("tracking-wasflight", wasFlying)
+            end
+        end
+        
+        self:SetData("tracking-ent",ent)
+        if IsValid(ent) and ent ~= wasTrackingEnt then
```

Comment:
we're not changing `ent` before, why not use isTracking here?

Reply:
https://github.com/MattJeanes/TARDIS/pull/969/commits/c5d1f4a007db37bb5c8a71e49e2a8ab9561b1e31